src/jdk.compiler/share/classes/com/sun/tools/javac/code/Symbol.java

```diff
@@ -1503,10 +1503,7 @@ public RecordComponent getRecordComponent(VarSymbol field) {
             return null;
         }
 
-        /* creates a record component if non is related to the given variable and recreates a brand new one
-         * in other case
-         */
-        public RecordComponent createRecordComponent(JCVariableDecl var, List<JCAnnotation> annotations) {
+        public RecordComponent findRecordComponentToRemove(JCVariableDecl var) {
```

Comment:
given that this method has only one client and it is simple, consider inlining its code in the client code

Reply:
`pos` has private access in `com.sun.tools.javac.code.Symbol.RecordComponent`. Is it ok if we bump it to `public` (it still hides the one from `VarSymbol`. WDYT?

Reply:
Oh I see then better to leave the code as it is now